WooCommerce/Classes/ViewRelated/Coupons/Add and Edit Coupons/AddEditCouponViewModel.swift

```diff
@@ -363,6 +434,8 @@ final class AddEditCouponViewModel: ObservableObject {
             return .couponCodeEmpty
         }
 
+        validatePercentageAmountInput(withWarning: false)
```

Comment:
I'm wondering if we should return an error when the percentage amount is invalid? From my testing, I found that the amount is automatically updated to 100 and then the coupon is updated right away. I'm not sure if that's a good UX or should we stop the update/creation when this happens?

If we choose to return an error, we can make `validatePercentageAmountInput` return an optional error, and the method can be marked with `@discardableResult` so that it's optional to use the returned error at call sites.

Reply:
Sorry, I'm not sure if I understood the scenario you described. Calling the `validatePercentageAmountInput` is causing a coupon update request right away? I can't verify in which scenario this would happen 🤔 Could you elaborate on how you reproduced that?

About the error handling, I wonder what exactly we would do with the amount field when we can't set a valid value there. The `validatePercentageAmountInput` function right now is adjusting the amount to zero in these scenarios:

<img width="832" alt="image" src="https://user-images.githubusercontent.com/5920403/176074263-71e5f2d0-3789-4d67-bf31-09128d2b8bc0.png">

 do you suggest handling this scenario differently when returning the error?

Reply:
I'm sorry for not including the screen recording earlier 😅 Here goes:


https://user-images.githubusercontent.com/5533851/176078154-1d49652f-acfe-41b3-8105-9bf668b2d99b.mp4

My suggestion was to stop the editing/creation flow if we find an issue with the percent amount. WDYT?